kore/src/Kore/Builtin/AssociativeCommutative.hs

```diff
@@ -807,254 +800,276 @@ unifyEqualsNormalized
             Normalized n -> return n
 
 {- | Unifies two AC structs represented as @NormalizedAc@.
-
 Currently allows at most one opaque term in the two arguments taken together.
 -}
-unifyEqualsNormalizedAc
-    ::  forall normalized unifier
-    .   ( Traversable (Value normalized)
-        , TermWrapper normalized
-        , MonadUnify unifier
-        )
-    => SmtMetadataTools Attribute.Symbol
-    -> TermLike RewritingVariableName
-    -> TermLike RewritingVariableName
-    ->  (  TermLike RewritingVariableName
-        -> TermLike RewritingVariableName
-        -> unifier (Pattern RewritingVariableName)
-        )
-    -> TermNormalizedAc normalized RewritingVariableName
-    -> TermNormalizedAc normalized RewritingVariableName
-    -> MaybeT
+unifyEqualsNormalizedAc ::
+    forall normalized unifier.
+    ( Traversable (Value normalized)
+    , TermWrapper normalized
+    , MonadUnify unifier
+    ) =>
+    TermLike RewritingVariableName ->
+    TermLike RewritingVariableName ->
+    ( TermLike RewritingVariableName ->
+      TermLike RewritingVariableName ->
+      unifier (Pattern RewritingVariableName)
+    ) ->
+    [Element normalized (TermLike RewritingVariableName)] ->
+    [Element normalized (TermLike RewritingVariableName)] ->
+    Map Key (Value normalized (TermLike RewritingVariableName)) ->
+    Map Key (Value normalized (TermLike RewritingVariableName)) ->
+    [TermLike RewritingVariableName] ->
+    [TermLike RewritingVariableName] ->
```

Comment:
Why are we taking these structures apart, instead of passing around the containers they came from? This is highly likely to introduce errors.